src/tup/Tupfile

```diff
@@ -1,3 +1,10 @@
 include_rules
 : foreach *.c | ../luabuiltin/luabuiltin.h |> !cc |>
-: foreach *.c | ../luabuiltin/luabuiltin.h |> !mingwcc |>
+ifneq (@(TUP_PLATFORM),win32)
+: |> version=`git describe` ; echo "const char *tup_version(void) {return \"$version\";}" >version.c |> version.c
```

Comment:
I left only this rule in for now, even on Windows. I do not see the error you describe when I run it on my Windows machine, and we'll want to have an actual version number built-in. Does it help if the commands are moved into a shell script and then the rule is invoked as ': |> bash generate-version.bash > %o |> version.c' or something similar?